Makefile

```diff
@@ -38,7 +38,7 @@ container: $(APP)
 push: container
 	docker push $(IMAGE_NAME):$(IMAGE_VERSION)
 
-test:
+test: $(APP)
```

Comment:
The commit message says "make test; make; make container", but that won't work anymore, right? It's perhaps better to put "make; make test; make container" into the commit message and mention in the Makefile for "test" that one has to do "make" first.

Reply:
I think the order does not really matter. All three `make`, `make test` and `make container` depend on `$(APP)`. So the first one executed compiles the binary, the others re-use it. You may notice I removed `make` in this PR, because it's useless.

Reply:
Right, it works here the way you said.

I was expecting "$(APP)" to be a phony target because there is no real dependency tracking for Go+make, i.e. "make $(APP)" always has to invoke "go build" to ensure that the binary really is up-to-date. But this Makefile here only builds the binary if it doesn't exist yet, without rebuilding it when it is out-dated.

I guess that's how it is meant to be here, so /lgtm.